src/mon/MgrMonitor.cc

```diff
@@ -588,6 +590,11 @@ bool MgrMonitor::prepare_beacon(MonOpRequestRef op)
       pending_map.clients = clients;
       updated = true;
     }
+  } else if (m->get_available()) {
+    dout(4) << "mgr thinks it is active but it is not, dropping!" << dendl;
+    auto m = make_message<MMgrMap>(MgrMap::create_null_mgrmap());
+    mon.send_reply(op, m.detach());
+    return true;
```

Comment:
Is `MMgrBeacon` message with `m->get_name() == pending_map.active_name && m->get_gid() != pending_map.active_gid && m->get_available()` possible?  If so, is returning without unplugging here intentional?

Reply:
> Is MMgrBeacon message with m->get_name() == pending_map.active_name && m->get_gid() != pending_map.active_gid && m->get_available() possible?

No, because the active is dropped if `m->get_name() == pending_map.active_name && m->get_gid() != pending_map.active_gid `. So, the names will not match:

https://github.com/ceph/ceph/blob/fe9eb488d0cbe83ae7fe01e809b1d07e8f14cccf/src/mon/MgrMonitor.cc#L508-L520

> If so, is returning without unplugging here intentional?

It is a code smell. I will fix it.